config/lib/websocket.js

```diff
@@ -0,0 +1,132 @@
+const connectMongo = require('connect-mongo');
+const cookieParser = require('cookie-parser');
+const http = require('http');
+const passport = require('passport');
+const path = require('path');
+const session = require('express-session');
+const WebSocket = require('ws');
+
+const config = require('../config');
+const MongoStore = connectMongo(session);
+
+function authenticateSession(req, connection, callback) {
+  // Create a MongoDB storage object
+  const mongoStore = new MongoStore({
+    collection: config.sessionCollection,
+    mongooseConnection: connection,
+  });
+
+  const cookieParserMiddleware = cookieParser(config.sessionSecret);
+
+  // Use the 'cookie-parser' module to parse the request cookies
+  cookieParserMiddleware(req, {}, err => {
+    if (err) {
+      console.error('[WebSocket] cookieParser error:', err);
+      return;
+    }
+
+    // Get the session id from the request cookies
+    const sessionId = req.signedCookies
+      ? req.signedCookies['connect.sid']
+      : undefined;
+
+    if (!sessionId) {
+      console.warn(
+        '[WebSocket] sessionId was not found in request',
+        req.signedCookies,
+      );
+      return;
+    }
+
+    // Use the mongoStorage instance to get the Express session information
+    mongoStore.get(sessionId, (err, session) => {
+      if (err) {
+        return console.error('[WebSocket] session error', err);
+      }
+
+      if (!session) {
+        console.warn(`[WebSocket] session was not found for ${sessionId}`);
+        return;
+      }
+
+      // Set the WebSocket session information
+      req.session = session;
+
+      // Use Passport to populate the user details
+      passport.initialize()(req, {}, () => {
+        passport.session()(req, {}, () => {
+          if (!req.user) {
+            return console.error(
+              new Error('[WebSocket] User is not authenticated'),
+            );
+          }
+
+          callback();
+        });
+      });
+    });
+  });
+}
+
+module.exports.start = function (app, connection) {
+  const map = new Map();
```

Comment:
would recommend using an Array instead to allow multiple connections (like web + app)

Reply:
Yup 👍 makes sense. Feel free to push changes to this branch!